src/misc/fs_utils_posix.cpp

```diff
@@ -457,4 +457,29 @@ DosDateTime get_dos_file_time(const NativeFileHandle handle)
 	return ret;
 }
 
+void set_dos_file_time(const NativeFileHandle handle, const uint16_t date, const uint16_t time)
+{
+	auto datetime = DOS_UnpackDateTime(date, time);
+
+	const auto unix_seconds = mktime(&datetime);
+	if (unix_seconds == -1) {
+		return;
+	}
+
+	struct timespec unix_times[2] = {};
+
+	// Last access time
+	// We don't really care about this but apparently it must be updated as well
+	unix_times[0].tv_sec = unix_seconds;
+	unix_times[0].tv_nsec = 0;
+
+	// Last modification time
+	unix_times[1].tv_sec = unix_seconds;
+	unix_times[1].tv_nsec = 0;
+
+	// I like hating on Win32 API but at least they have better function names...
```

Comment:
💯 

The Win32 API is quite nice.

Reply:
Please fill out this struct with no less than 10 fields describing how much you like the Win32 API.

Reply:
To their credit, the struct approach is superior for C vs doing a function call like this:

```c
do_something_maybe(true, false, 0, 0, 4443, NULL, false, 3);
```

The structs are the poor man's keyword args for C & C++ 😆